### PR TITLE
Save files into folders

### PR DESCRIPTION
**Pull request by denar90**
I've been thinking, if project is big one folder will consist a lot of files. Maybe, saving files fix this problem.

### CODE REVIEW

File: actions/UploadAction.php [hunk 1]
```diff
@@ -110,6 +110,12 @@ public function init()
     public function run()
     {
         if (Yii::$app->request->isPost) {
+            $imageDir = '/' . date('d-m-Y') . '/';
```
Comment (vova07) [inline actions/UploadAction.php]: - Will be better to have here a function call to can generate own rule for directory name
- Need a option to turn on/off this behavior.

File: actions/UploadAction.php [hunk 2]
```diff
@@ -110,6 +110,12 @@ public function init()
...
+            $this->path = $this->path . $imageDir;
```
Comment (vova07) [inline actions/UploadAction.php]: Need to add `DIRECTORY_SEPARATOR` at the end of this line.

File: actions/UploadAction.php [hunk 3]
```diff
@@ -110,6 +110,12 @@ public function init()
...
+            $this->url = $this->url . $imageDir;
```
Comment (vova07) [inline actions/UploadAction.php]: Need to add `/` at the end of this line.

File: actions/UploadAction.php [hunk 4]
```diff
@@ -110,6 +110,12 @@ public function init()
...
+            if (!is_dir($this->path)) {
+                mkdir($this->path, 0777, true);
+            }
```
Comment (vova07) [inline actions/UploadAction.php]: I think `FileHelper::createDirectory($this->path)` is better here.

**Comment by vova07** [comment]
All this manipulation can be done in `init()` method. Will be nicer.

**Comment by denar90** [comment]
A've added changes. Maybe it is make sense return also filepath in response?

**Comment by vova07** [comment]
Is this really necessary to add one more param for directory? I mean, if this `directory` will be setted from outside, why to not use the same `path` and `url` and to do the concatenation manually?

**Comment by denar90** [comment]
As I've got your point, you don't want to add new param $fileDirectory, am I wright?
As I've being thinking, for example, I can set directory name as "date('d-m-Y')". It will create new folders. And for new day, due to example, new folder for images will be created.

**Comment by vova07** [comment]
yes, I get your point of view. But what I want to say is, that this case can be solve really easy with the current code. Real example:
```php
// Add images that have already been uploaded

// DefaultController.php
public function actions()
{
    return [
        'images-get' => [
            'class' => 'vova07\imperavi\actions\GetAction',
            'url' => 'http://my-site.com/images/' . date('d-m-Y') . '/', // So we just write the right URL from the start.
            'path' => '@alias/to/my/path' . date('d-m-Y') . '/', // Same thing as with URL
            'type' => GetAction::TYPE_IMAGES,
        ]
    ];
}

// View.php
echo \vova07\imperavi\Widget::widget([
    'selector' => '#my-textarea-id',
    'settings' => [
        'lang' => 'ru',
        'minHeight' => 200,
        'imageManagerJson' => Url::to(['/default/images-get']),
        'plugins' => [
            'imagemanager'
        ]
    ]
]);
```

**Comment by denar90** [comment]
Good point. You can delete this pull request.

**Pull request closed without merging**